### PR TITLE
#145 improve YT links LHS via ingesting lite-youtube-embed

### LINKED ISSUE #145: LHS: use  lite-youtube-embed for youtube video embeds
We experienced Low LHS on certain pages that contain embedded YouTube videos.
The problem was associated with the Embed Block, specifically with the script Embed Player .
This issue can arise if a YouTube video is present in the above-the-fold content.

We can use third party facades to tackle this as recommended on [google developers page](https://developer.chrome.com/docs/lighthouse/performance/third-party-facades/).
These facades don't load complex JS/CSS beforehand but after clicking on video they loads.

we can use https://github.com/paulirish/lite-youtube-embed - also used in other Franklin projects

### PR DESCRIPTION
**Pull request by meejain**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #145 

Test URLs:
- Before: https://main--24life--hlxsites.hlx.page/magazine/volume-6-issue-1
- After: https://issue145-improvelhs-ytvideos--24life--hlxsites.hlx.page/drafts/meejain/magazine/volume-6-issue-1-v2
- https://issue145-improvelhs--24life--hlxsites.hlx.page/block-library/blocks/youtube

### CODE REVIEW

File: blocks/embed/lite-yt-embed.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
+/**
```
Comment (synox) [inline blocks/embed/lite-yt-embed.js:1]: > Aravindo Wingeier  [1 day ago] 
> we should copy as is, add a comment with the source url, and disable linting for that file with a comment

Comment (meejain) [reply blocks/embed/lite-yt-embed.js:2]: Fixed

Comment (synox) [reply blocks/embed/lite-yt-embed.js:2]: >add a comment with the source url

File: scripts/scripts.js [hunk 2]
```diff
@@ -107,20 +108,58 @@ function decorateVideoLinks(main) {
...
+        } else if (autoplayParam) {
+          suffix += `&autoplay=${autoplayParam}&muted=1`;
+        } else if (mutedParam) {
+          suffix += `&muted=${mutedParam}`;
+        }
+        const vid = youtubeVideoId;
+        let embedHTML = '';
+        embedHTML = `
+      <lite-youtube videoid=${vid || embedSplit[embedSplit.length - 1]}>
+        <a href="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&v=${vid}${suffix}` : embed}" class="lty-playbtn" title="Play Video">
```
Comment (synox) [inline scripts/scripts.js:156]: can we use `https://www.youtube-nocookie.com` like the previous code?

File: scripts/scripts.js [hunk 3]
```diff
@@ -94,23 +105,15 @@ function decorateVideoLinks(main) {
...
-        iframe.title = 'YouTube video player';
-        iframe.frameborder = 0;
-        iframe.allow = 'accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture;';
-        iframe.allowfullscreen = true;
-        link.replaceWith(iframe);
+        loadCSS(`${window.hlx.codeBasePath}/blocks/embed/lite-yt-embed.css`);
+        loadScript(`${window.hlx.codeBasePath}/blocks/embed/lite-yt-embed.js`);
+        const ltYoutube = document.createElement('lite-youtube');
+        ltYoutube.setAttribute('videoid', youtubeVideoId);
+        ltYoutube.setAttribute('params', 'rel=0');
```
Comment (synox) [inline scripts/scripts.js:115]: The play button opened the video full screen. removing it solved it.

**Review by synox — APPROVED** [review]

**Pull request merged**